pennylane/templates/subroutines/select.py

```diff
@@ -723,72 +733,338 @@ def _add_k_units(ops, controls, work_wires, k):
     )
 
 
-# pylint: disable=unused-argument
-def _select_resources_partial_unary(op_reps, num_control_wires, partial):
-    if not partial:
-        raise NotImplementedError(
-            "Resources for unary iteration with partial=False not implemented yet."
+def _select_resources_unary_not_partial(op_reps, num_control_wires, num_work_wires):
+    resources = defaultdict(int)
+    c = num_control_wires
+    K = len(op_reps)
+    num_work_wires = num_work_wires - (c - 1)
+
+    if c == 1:
+        for i, target_rep in enumerate(op_reps):
+            resources[
+                controlled_resource_rep(
+                    base_class=target_rep.op_type,
+                    base_params=target_rep.params,
+                    num_control_wires=1,
+                    num_zero_control_values=(1 - i),
+                    num_work_wires=num_work_wires,
+                )
+            ] += 1
+        return dict(resources)
+
+    def _make_first_flipped_bits(c, i=0):
```

Comment:
tbh I didnt fully grasp this, could you add a little more description for other devs what this function is doing, and how its doing that?

Reply:
recursion + walrus is hard to read/follow

Reply:
Removed the walrus and added an example to the docstring.